### ISSUE TITLE
Unexpected com.google.api.gax.rpc.InvalidArgumentException

### ISSUE THREAD

**Issue by christopherL91** [GoogleCloudPlatform/opentelemetry-operations-java#46]
Hi!

I get a strange exception thrown after I've loaded the auto jar, I can see new traces in cloud trace but after a few seconds this pops up:

```
com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: One or more TimeSeries could not be written: Unrecognized resource type: .: timeSeries[0]
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:49)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:72)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:60)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:68)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1039)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1165)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:958)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:749)
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:545)
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:515)
	at io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23)
	at io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40)
	at io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1$1.onClose(CensusStatsModule.java:700)
	at io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39)
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23)
	at io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40)
	at io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1$1.onClose(CensusTracingModule.java:399)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:510)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:66)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:630)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:518)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:692)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:681)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
	Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
		at com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
		at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
		at com.google.cloud.monitoring.v3.MetricServiceClient.createTimeSeries(MetricServiceClient.java:1146)
		at com.google.cloud.monitoring.v3.MetricServiceClient.createTimeSeries(MetricServiceClient.java:1088)
		at com.google.cloud.opentelemetry.metric.CloudMetricClientImpl.createTimeSeries(CloudMetricClientImpl.java:24)
		at com.google.cloud.opentelemetry.metric.MetricExporter.createTimeSeriesBatch(MetricExporter.java:159)
		at com.google.cloud.opentelemetry.metric.MetricExporter.export(MetricExporter.java:148)
		at io.opentelemetry.sdk.metrics.export.IntervalMetricReader$Exporter.run(IntervalMetricReader.java:196)
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
		at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
		at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
		... 3 more
Caused by: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: One or more TimeSeries could not be written: Unrecognized resource type: .: timeSeries[0]
	at io.grpc.Status.asRuntimeException(Status.java:533)
	... 23 more
```

I've seen a few similar logs posted online but with no solution so far.

**Comment by jsuereth** [comment]
Sorry for taking so long to look at this.  Can you share the code you have for pushing a custom metric?

The error message is saying that it's missing a `resource_type` (for some reason it's trying to push an empty string as the resource).

**Comment by tomcham** [comment]
@jsuereth Also getting the same exception. Sample code running in GKE:
```
  val interval =
    IntervalMetricReader
      .builder()
      .setMetricProducers(Collections.singleton(OpenTelemetrySdk.getGlobalMeterProvider().getMetricProducer()))
      .setExportIntervalMillis(60 * 1000)
      .setMetricExporter(
        MetricExporter
          .createWithConfiguration(MetricConfiguration.builder().setProjectId("xxxx").build())
      )
      .build()

  val meter   = OpenTelemetry.getGlobalMeterProvider.get("test", "1.0")
  val counter = meter.longCounterBuilder("test_counter").build()
  counter.add(2, Labels.of("label", "label-value"))
  interval.shutdown()
```

**Comment by johndoe8224** [comment]
I think this issue is from https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/v0.10.x/exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricExporter.java#L144

**Comment by jsuereth** [comment]
0.14.0-RC1 is out (against OTel Java 0.14.1).  @tomcham Would you mind giving this a try and seeing if it fixes things?  I'd like to make sure we have this fixed before the 0.14.0 release if we can.

**Comment by tomcham** [comment]
> 0.14.0-RC1 is out (against OTel Java 0.14.1). @tomcham Would you mind giving this a try and seeing if it fixes things? I'd like to make sure we have this fixed before the 0.14.0 release if we can.

It's working now.

**Comment by aditya-protonn** [comment]
What's the closure on this ? Unable to get rid of this error. I was using google transcoder jar and faced this  : 

```
13:49:03.270 [grpc-nio-worker-ELG-1-4] DEBUG io.grpc.netty.shaded.io.grpc.netty.NettyClientHandler - [id: 0x753448b2, L:/192.168.43.63:52035 - R:transcoder.googleapis.com/142.250.192.106:443] OUTBOUND PING: ack=true bytes=0
Exception in thread "main" com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Request contains an invalid argument.
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:49)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:72)
	at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:60)
	at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
	at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:68)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1041)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1215)
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:983)
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:771)
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:563)
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:533)
	at io.grpc.internal.DelayedClientCall$DelayedListener$3.run(DelayedClientCall.java:464)
	at io.grpc.internal.DelayedClientCall$DelayedListener.delayOrExecute(DelayedClientCall.java:428)
	at io.grpc.internal.DelayedClientCall$DelayedListener.onClose(DelayedClientCall.java:461)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:617)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:803)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:782)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
	Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
		at com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
		at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
		at com.google.cloud.video.transcoder.v1beta1.TranscoderServiceClient.createJob(TranscoderServiceClient.java:237)
		at com.protonn.googletranscoder.CreateJobWithStaticOverlay.createJobWithStaticOverlay(CreateJobWithStaticOverlay.java:125)
		at com.protonn.googletranscoder.CreateJobWithStaticOverlay.main(CreateJobWithStaticOverlay.java:28)
Caused by: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Request contains an invalid argument.
	at io.grpc.Status.asRuntimeException(Status.java:533)
	... 16 more
13:49:03.301 [grpc-nio-worker-ELG-1-4] DEBUG io.grpc.netty.shaded.io.grpc.netty.NettyClientHandler - [id: 0x753448b2, L:/192.168.43.63:52035 - R:transcoder.googleapis.com/142.250.192.106:443] OUTBOUND GO_AWAY: lastStreamId=0 errorCode=0 length=0 bytes=

> Task :CreateJobWithStaticOverlay.main() FAILED
```

**Comment by jsuereth** [comment]
@aditya-protonn That error looks like it's from `com.google.cloud.video.transcoder.v1beta1.TranscoderServiceClient`, so should be unrelated to this project.

This bug should be fixed as of 0.14 ( and 0.15 release).
